kahuna/public/js/edits/service.js

```diff
@@ -117,12 +117,14 @@ service.factory('editsService',
     // update requests for a given Resource
     const updateRequestPools = new Map();
 
-    function registerUpdateRequest(resource, originalImage) {
+    function registerUpdateRequest(resource, originalImage, inBatch = false) {
         const requestPool = createRequestPool();
         const promise = withWatcher(resource, requestPool.promise).
               then(({ edit, image }) => {
-                  $rootScope.$emit('image-updated', image, originalImage);
-                  return edit;
+                if (!inBatch) {
+                  $rootScope.$emit('images-updated', [image]);
```

Comment:
To confirm my understanding – the idea here is that if this caller knows it's not being called as a part of a batch, it sends an event itself; if it _is_ called as part of a batch, it delegates to the caller, which presumably is sending the event itself?

If that's correct, be good to document in the function description.

Reply:
Yeah, where I could I pulled the functions apart properly. But `service.js` is too widely used and the implementation is far too clever to attempt refactoring- so I just bunged in an optional `inBatch` parameter.